src/app/rpmostree-builtin-status.c

```diff
@@ -705,14 +705,15 @@ print_one_deployment (RPMOSTreeSysroot *sysroot_proxy,
           const char *name, *nevra;
           g_variant_get_child (gv_nevra, 0, "&s", &nevra);
           g_variant_get_child (gv_nevra, 1, "&s", &name);
-          if (str->len)
-            g_string_append (str, ", ");
-          g_string_append (str, nevra);
           g_ptr_array_add (active_removals, g_strdup (name));
+          g_ptr_array_add (active_removals_nevra, g_strdup (nevra));
         }
       g_ptr_array_add (active_removals, NULL);
-      if (str->len)
-        rpmostree_print_kv ("RemovedBasePackages", max_key_len, str->str);
+      g_ptr_array_add (active_removals_nevra, NULL);
+
+      if (active_removals_nevra)
```

Comment:
Ahh right, this is a gotcha right here. So, we only want to print it if there's active removals. But the pointer will always be non-`NULL` so this will always execute right now. So we'd want something like this:

```c
if (active_removals_nevra->len > 0)
```

But that has another gotcha, which is that it will also always be true because we always add a final `NULL` element just above this line. So what we can do here is to only add the final `NULL` if there's elements to print, i.e.:

```c
if (active_removals_nevra->len > 0)
  {
    g_ptr_array_add (active_removals_nevra, NULL);
    print_packages (...);
  }
```

Reply:
Good catch!  `print_packages` checks the length of pointer and returns if the length is zero so I missed this in testing.